neevaco/neeva-ios
# Resolve tab parent UUID when restoring from SavedTab

Changes:
- Add routine to resolve parentUUID to parent refs
- add calls to above routine in `TabManager.restoreSavedTabs` and `TabManagerStore.restoreTabs`

also fix setting bad values to `tab.parentUUID` and `tab.tabUUID` in `SavedTab.configureTab`

## Checklists

### How was this tested?

### Manual test cases
1) create tabs with parent tabs (from about:blank or cheatsheet) -> close app -> reopen app -> verify back button to parent tab behaviours still work. This tests parent tabs are restored on startup in `TabManagerStore.restoreTabs`
2) create tabs with parent tabs -> go to tab switcher, close the tab -> tab restore button -> verify back button. This tests parent tab is restored in `TabManager.restoreSavedTabs`

### Screenshots and screen recordings

## Issues addressed
